services/blockchain-service/blockchain-service-base.js

```diff
@@ -137,22 +137,41 @@ class BlockchainServiceBase {
         return this[blockchain.name].contracts[blockchain.hubContract][contractName];
     }
 
-    async createAsset(requestData, blockchain, stepHooks = emptyHooks) {
+    async createAsset(
+        requestData,
+        blockchain,
+        stepHooks = emptyHooks,
+        skipIncreaseAllowance = false,
+    ) {
         const serviceAgreementV1Address = await this.getContractAddress(
             'ServiceAgreementV1',
             blockchain,
         );
 
-        await this.executeContractFunction(
-            'Token',
-            'increaseAllowance',
-            [serviceAgreementV1Address, requestData.tokenAmount],
-            blockchain,
-        );
+        if (!skipIncreaseAllowance) {
+            const allowance = await this.callContractFunction(
+                'Token',
+                'allowance',
+                [blockchain.publicKey, serviceAgreementV1Address],
+                blockchain,
+            );
 
-        stepHooks.afterHook({
-            status: OPERATIONS_STEP_STATUS.INCREASE_ALLOWANCE_COMPLETED,
-        });
+            if (BigInt(allowance) < BigInt(requestData.tokenAmount)) {
+                await this.executeContractFunction(
+                    'Token',
+                    'increaseAllowance',
+                    [
+                        serviceAgreementV1Address,
+                        BigInt(requestData.tokenAmount) - BigInt(allowance),
```

Comment:
please avoid doing same calculations in multiple places (conversion and subtraction)